plugins/zify/g_zify.mlg

```diff
@@ -0,0 +1,77 @@
+(************************************************************************)
+(*         *   The Coq Proof Assistant / The Coq Development Team       *)
+(*  v      *   INRIA, CNRS and contributors - Copyright 1999-2019       *)
+(* <O___,, *       (see CREDITS file for the list of authors)           *)
+(*   \VV/  **************************************************************)
+(*    //   *    This file is distributed under the terms of the         *)
+(*         *     GNU Lesser General Public License Version 2.1          *)
+(*         *     (see LICENSE file for the text of the license)         *)
+(************************************************************************)
+
+{
+
+open Ltac_plugin
+open Stdarg
+open Tacarg
+open Pp
+
+let iter_hyps (tac:Tacinterp.Value.t) =
+  Proofview.Goal.enter  begin fun gl ->
+    let hyps = List.rev (Tacmach.New.pf_hyps_types gl) in
+
+    Tacticals.New.tclTHENLIST
+          (List.map (fun (h,t) -> Tacticals.New.tclTRY (Tacinterp.Value.apply tac [Tacinterp.Value.of_constr (EConstr.mkVar h) ; Tacinterp.Value.of_constr t])) hyps)
+    end
+
+let iter_specs (tac: Tacinterp.Value.t) =
+    Zify.iter_specs (fun c -> Tacinterp.Value.apply tac [Tacinterp.Value.of_constr c])
+
+}
+
+
+
+DECLARE PLUGIN "zify_plugin"
+
+
+VERNAC COMMAND EXTEND DECLAREINJECTION CLASSIFIED AS SIDEFF
+| ["Add" "InjTyp"    constr(t) ] -> { Zify.InjTable.register t }
```

Comment:
These commands take a `constr` arguments, but they seem used solely on names. Couldn't you simply reuse the `Coqlib` machinery here, instead of defining a new registration mechanism?

Reply:
Would that fit the bill? The registration is performing some work e.g. initialise properly indexed tables.